weld/subsystem/src/main/java/org/jboss/as/weld/WeldResourceDefinition.java

```diff
@@ -91,7 +92,7 @@ class WeldResourceDefinition extends PersistentResourceDefinition {
             .setRestartAllServices()
             .build();
 
-    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] { REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE, LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE, NON_PORTABLE_MODE_ATTRIBUTE, DEVELOPMENT_MODE_ATTRIBUTE, THREAD_POOL_SIZE_ATTRIBUTE };
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] { REQUIRE_BEAN_DESCRIPTOR_ATTRIBUTE, LEGACY_EMPTY_BEANS_XML_TREATMENT_ATTRIBUTE, NON_PORTABLE_MODE_ATTRIBUTE, THREAD_POOL_SIZE_ATTRIBUTE };
```

Comment:
This attribute can't be dropped as it's needed to support mixed version domain with secondary hosts running earlier versions. Also before removing we need to deprecate it and warn (see .setDeprecated... bit above) to give users a chance to remove it from any CLI scripts etc.  We don't have to retain the behavior; we just need to keep the attribute and make it a no-op other than the warn.